File: templates/problems/07_trees/06_subtree/go/subtree_test.go
package subtree

import "testing"

func TestIsSubtree(t *testing.T) {
	tests := []struct {
		name    string
		root    []int
		subRoot []int
		want    bool
	}{
		{"is subtree", []int{3, 4, 5, 1, 2}, []int{4, 1, 2}, true},
		{"not subtree", []int{3, 4, 5, 1, 2, -101, -101, -101, -101, 0}, []int{4, 1, 2}, false},
		{"both single same", []int{1}, []int{1}, true},
		{"both single diff", []int{1}, []int{2}, false},
		{"sub is nil", []int{1, 2, 3}, []int{}, true},
		{"root equals sub", []int{1, 2, 3}, []int{1, 2, 3}, true},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			root := buildTreeHelper(tt.root)
			subRoot := buildTreeHelper(tt.subRoot)
			if got := isSubtree(root, subRoot); got != tt.want {
				t.Errorf("isSubtree() = %v, want %v", got, tt.want)
			}
		})
	}
}

File: templates/problems/07_trees/06_subtree/typescript/subtree.test.ts
import { describe, it, expect } from "vitest";
import { buildTree } from "./helpers";
import { isSubtree } from "./subtree";

describe("isSubtree", () => {
    it.each([
        { name: "is subtree", root: [3, 4, 5, 1, 2], subRoot: [4, 1, 2], want: true },
        { name: "not subtree", root: [3, 4, 5, 1, 2, -101, -101, -101, -101, 0], subRoot: [4, 1, 2], want: false },
        { name: "both single same", root: [1], subRoot: [1], want: true },
        { name: "both single diff", root: [1], subRoot: [2], want: false },
        { name: "sub is nil", root: [1, 2, 3], subRoot: [], want: true },
        { name: "root equals sub", root: [1, 2, 3], subRoot: [1, 2, 3], want: true },
    ])("case: $name", ({ root, subRoot, want }) => {
        expect(isSubtree(buildTree(root), buildTree(subRoot))).toBe(want);
    });
});
